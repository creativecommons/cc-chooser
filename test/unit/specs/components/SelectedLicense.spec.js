import { shallowMount } from '@vue/test-utils'
import SelectedLicense from '@/components/SelectedLicense.vue'

describe('SelectedLicense.spec.js', () => {
    let cmp

    beforeEach(() => {
        cmp = shallowMount(SelectedLicense, {
        // Be aware that props is overridden using `propsData`
            propsData: {
                fullLicenseName: 'Attribution',
                shortLicenseName: 'BY'
            }
        })
    })

    it('has received \'Attribution\' as the fullLicenseName property', () => {
        expect(cmp.vm.fullLicenseName).toEqual('Attribution')
    })

    it('has received \'BY\' as the shortLicenseName property', () => {
        expect(cmp.vm.shortLicenseName).toEqual('BY')
    })

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot()
    })
})
