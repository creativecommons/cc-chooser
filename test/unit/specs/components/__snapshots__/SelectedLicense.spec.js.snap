// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedLicense.spec.js has the expected html structure 1`] = `
<div
  id="SelectedLicense"
>
  <div
    class="card chooser-selected"
  >
    <header
      class="card-header"
    >
      <h2
        class="title is-2"
      >
        Selected License
      </h2>
    </header>
     
    <div
      class="card-content"
    >
      <div>
        <licenseiconography-stub
          iconlist=","
        />
         
        <div
          class="selected-license-names"
        >
          <b
            class="selected-license-name"
          >
            BY
          </b>
           
          <a
            class="help"
            href="https://creativecommons.org/licenses//4.0/?ref=ccchooser"
          >
            Attribution
          </a>
        </div>
      </div>
       
      <licensedescription-stub
        selectedlicense="BY"
      />
    </div>
  </div>
</div>
`;
